File: 2-AddTwoNumbers/python/src/add_two_numbers_1.py
# Definition for singly-linked list.
class ListNode:
     def __init__(self, val=0, next=None):
         self.val = val
         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        maxLength = 0
        result = []
        
        # if len(l1) > len(l2):
        #     maxLength = len(l2)
        # elif len(l2) < len(l1):
        #     maxLength = len(l1)
        # else:
        #     maxLength = len(l1)
            
        carry = 0
        
        for (a, b, c) in zip(l1, l2):
            
        
        return result                

File: 2-AddTwoNumbers/csharp/src/add_two_numbers_1.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode AddTwoNumbers(ListNode l1, ListNode l2) {
        ListNode sumListHead = null;
        ListNode sumListIndex = null;
        ListNode nextNode = null;
        int carry = 0;
              
        while(l1 != null || l2 != null) {
            int value1, value2 = 0;
            if(l1 == null) {
                value1 = 0;
                value2 = l2.val;
                l2 = l2.next;
            } else if(l2 == null) {
                value1 = l1.val;
                value2 = 0;
                l1 = l1.next;
            } else {
                value1 = l1.val;
                value2 = l2.val;
                l1 = l1.next;
                l2 = l2.next;
            }
                            
            nextNode = new ListNode((value1 + value2 + carry) % 10);
                
            if (value1 + value2 + carry >= 10)
                    carry = 1;
                else
                    carry = 0;